jukejoint_frontend/src/pod/player/components/PlayerDuration.tsx

```diff
@@ -1,16 +1,112 @@
 import React, { Component } from 'react';
+import { AppContextProp, withAppContext, AppContextType } from '../../../toolkit/util/AppContext';
 import styles from '../styles/PlayerDuration.module.css';
+import { debug } from 'util';
+import { SongModel } from '@jukejoint/common/lib/models';
+
+interface IProps extends AppContextProp {
+  // interval: NodeJS.Timer
+  // currentCount: number;
+  appContext: AppContextType;
+}
+
+interface IState {
+  currentTime: number;
+  songDuration: number;
+}
+
+class SongDuration extends Component<IProps, IState> {
+  private interval?: NodeJS.Timeout;
+  
+  constructor(props:any) {
+    super(props);
+    this.state = {
+      currentTime: 0,
+      songDuration: 0
+    }
+  }
 
-class SongDuration extends Component {
+
+
+  componentWillUnmount() {
+    console.log("Unmounting PlayerDuration...");
+    clearInterval(this.interval!);
+  }
+
+  shouldComponentUpdate(nextProps: Readonly<IProps>, nextState: Readonly<IState>, nextContext: any): boolean {
```

Comment:
I think there are a few things wrong that can be improved here.
1. We can remove all the `player` and `player.currentSong` sanity checks as they are already checked in the parent or can be, no need to recheck.
2. For resetting timer we can simply check whether the song id of current and next song are same or not.
3. For pausing and restarting the timer on song pause and play, we just need to see the `isPlaying` state of the player.

Reply:
Regardings points #2 and #3: if we remove `appContext` from the props, we'd also lose access to the `player`, and with it `player.currentSong` which we need for the `id` and `duration` properties, as well as `player.isPlaying`, which may be solved by passing all from the parent. Should I pass these props from the parent?

Reply:
Not sure what you mean, that is essentially (1). Yup, we should pass those as props from parents. But what do you think about (2) and (3)?

P.S. hashtags in github would make a reference to issues or PRs, like in this case you just referenced some PRs :)

Reply:
I'll pass `currentSong` and `isPlaying` from parent